mergify_engine/rules/__init__.py

```diff
@@ -289,18 +397,28 @@ async def create(
 
             await rule.conditions.eval(ctxt.pull_request)
 
-            for condition in rule.conditions:
-                if hide_rule and condition.evaluation_error:
-                    self.faulty_rules.append(typing.cast(T_EvaluatedRule, rule))
-                    break
-                elif (
-                    hide_rule
-                    and condition.get_attribute_name() in self.BASE_ATTRIBUTES
-                    and not condition.match
-                ):
+            categorized = False
+
+            if hide_rule and not rule.conditions.match:
+                # NOTE(sileht): Replace non base attributes by true, if it
+                # still match it's a potential rule otherwise hide it.
```

Comment:
```suggestion
                # NOTE(sileht): Replace non-base attribute by true, if it
                # still matches it's a potential rule otherwise hide it.
```